cloudferrylib/os/identity/keystone.py

```diff
@@ -164,7 +164,7 @@ def read_info(self, **kwargs):
         if kwargs.get('tenant_id'):
             self.filter_tenant_id = kwargs['tenant_id'][0]
 
-        tenant_list = self.get_tenants_list()
+        tenant_list = self.get_tenants_list(self.filter_tenant_id)
```

Comment:
Passing instance member (`self.filter_tenant_id`) as argument to instance method (`self.get_tenants_list`) looks VERY strange. This smells bad code, please refactor.